Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -559,6 +560,36 @@ public int handleManaXMultiplier(Game game, int value) {
         return xEvent.getAmount();
     }
 
+    /**
+     * Handles X mana costs where each mana has to be a different color (Emblazoned Golem)
+     */
+    private String handleManaXDifferentCost(Game game, Player controller, int amountMana) {
+        if (amountMana >= 5) {
+            return("{W}{U}{B}{R}{G}");
+        }
+
+        // Choose a different color of mana X times
+        ArrayList<ChoiceColor> chosenColors = new ArrayList<>();
+        for (int i = 0; i < amountMana; i++) {
+            ChoiceColor choice = new ChoiceColor();
+            for (ChoiceColor alreadyChosen : chosenColors) {
+                choice.removeColorFromChoices(alreadyChosen.getChoice());
+            }
+
+            if (controller.choose(Outcome.Benefit, choice, game)) {
+                chosenColors.add(choice);
+            }
+        }
```

Comment:
Check out https://github.com/magefree/mage/pull/7528. Might be a more elegant solution. Let them pay a max of 5 (and limit each color to 1).

That limit each color to 1 may need a change to the picker.

Reply:
This is a good suggestion! Changing the dropdown to checkboxes might be the better solution here than mine.

However, i'm currently unable to do UI changes with my current Intellij setup, so this will probably have to do.

Reply:
You don't need access to the UI builder for this part. To bring up the window you don't need any to make any changes to it, take a look at how one of the cards mentioned in that PR/associate issues does this.

As for limiting it to one of each color, you have two options:
1. Best, but requires changes to the picker (not the ui part of it): ask for max of 5 mana limiting the UI to only allow 1 of each color.
2. Simpler: ask for a max of 5, and then filter the Mana result to cap each color to a max of 1.